src/renderer/components/archive/entries-list.js

```diff
@@ -54,13 +56,25 @@ ListItemContent.propTypes = {
   entry: PropTypes.object
 };
 
-const List = ({ entries, currentEntry, onSelectEntry, onRightClick }) => (
+const List = ({
+  entries,
+  currentEntry,
+  currentEntryMode,
+  onSelectEntry,
+  onRightClick
+}) => (
   <ListWrapper>
     <For each="entry" of={entries}>
       <ListItem
         key={entry.id}
         active={currentEntry && entry.id === currentEntry.id}
-        onClick={() => onSelectEntry(entry.id)}
+        onClick={() =>
+          !currentEntry && currentEntryMode === 'new'
+            ? showConfirmDialog(
```

Comment:
This should be handled in the redux actions file. Search the project for `showConfirmDialog` usages to see how it's done in other places. It should not be handled inside the presentation components.